File: app/src/main/kotlin/com/apex/trader/presentation/screen/setup/SetupScreen.kt
package com.apex.trader.presentation.screen.setup

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material3.Button
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Switch
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.apex.trader.presentation.component.ApexCard
import com.apex.trader.presentation.theme.ApexHighlight
import com.apex.trader.presentation.theme.ApexTextMuted

@OptIn(androidx.compose.material3.ExperimentalMaterial3Api::class)
@Composable
fun SetupScreen(
    onSaved: () -> Unit,
    vm: SetupViewModel = hiltViewModel()
) {
    var apiKey by remember { mutableStateOf("") }
    var apiSecret by remember { mutableStateOf("") }
    var testnet by remember { mutableStateOf(false) }
    val ui by vm.ui.collectAsStateWithLifecycle()

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Connect Binance") },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = MaterialTheme.colorScheme.background
                )
            )
        }
    ) { padding ->
        Column(
            modifier = Modifier.fillMaxSize().padding(padding).padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            ApexCard {
                Column {
                    Text("Apex Trader", style = MaterialTheme.typography.headlineMedium)
                    Spacer(Modifier.height(4.dp))
                    Text(
                        "Multi-timeframe confluence scanner & trade executor for Binance USDT-M Futures.",
                        color = ApexTextMuted,
                        style = MaterialTheme.typography.bodyMedium
                    )
                    Spacer(Modifier.height(12.dp))
                    Text(
                        "Use a Futures-only API key. Disable withdrawals; restrict to your IP. Trading risk is yours.",
                        color = ApexHighlight,
                        style = MaterialTheme.typography.bodyMedium
                    )
                }
            }

            OutlinedTextField(
                value = apiKey,
                onValueChange = { apiKey = it.trim() },
                label = { Text("API Key") },
                singleLine = true,
                modifier = Modifier.fillMaxWidth()
            )
            OutlinedTextField(
                value = apiSecret,
                onValueChange = { apiSecret = it.trim() },
                label = { Text("API Secret") },
                singleLine = true,
                visualTransformation = PasswordVisualTransformation(),
                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Password),
                modifier = Modifier.fillMaxWidth()
            )
            Row(verticalAlignment = Alignment.CenterVertically) {
                Switch(checked = testnet, onCheckedChange = { testnet = it })
                Spacer(Modifier.height(8.dp))
                Text("  Use Testnet", color = ApexTextMuted)
            }

            if (ui.error != null) {
                Text(ui.error!!, color = MaterialTheme.colorScheme.error)
            }

            Button(
                onClick = { vm.save(apiKey, apiSecret, testnet, onSaved) },
                enabled = !ui.saving && apiKey.isNotBlank() && apiSecret.isNotBlank(),
                modifier = Modifier.fillMaxWidth()
            ) {
                Text(if (ui.saving) "Verifying…" else "Verify & continue")
            }
        }
    }
}

File: lib/features/setup/setup_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../core/theme.dart';
import '../../data/local/secure_credential_store.dart';
import '../../providers.dart';
import '../../widgets/common.dart';

class SetupScreen extends ConsumerStatefulWidget {
  const SetupScreen({super.key, required this.onSaved});
  final VoidCallback onSaved;

  @override
  ConsumerState<SetupScreen> createState() => _SetupScreenState();
}

class _SetupScreenState extends ConsumerState<SetupScreen> {
  final _key = TextEditingController();
  final _secret = TextEditingController();
  bool _testnet = false;
  bool _saving = false;
  String? _error;

  @override
  void dispose() {
    _key.dispose();
    _secret.dispose();
    super.dispose();
  }

  Future<void> _save() async {
    setState(() {
      _saving = true;
      _error = null;
    });
    final store = ref.read(credentialsStoreProvider);
    final api = ref.read(binanceApiProvider);
    await store.save(BinanceCredentials(
      apiKey: _key.text.trim(),
      apiSecret: _secret.text.trim(),
      testnet: _testnet,
    ));
    try {
      // Verify by fetching the account; if it fails, roll back creds.
      await api.getAccount();
      if (!mounted) return;
      setState(() => _saving = false);
      widget.onSaved();
    } catch (e) {
      await store.clear();
      if (!mounted) return;
      setState(() {
        _saving = false;
        _error = 'Authentication failed: ${e.toString()}';
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    final canSave = !_saving && _key.text.trim().isNotEmpty && _secret.text.trim().isNotEmpty;
    return Scaffold(
      appBar: AppBar(title: const Text('Connect Binance')),
      body: ListView(
        padding: const EdgeInsets.all(16),
        children: [
          ApexCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text('Apex Trader', style: Theme.of(context).textTheme.headlineSmall),
                const SizedBox(height: 4),
                const Text(
                  'Multi-timeframe confluence scanner & trade executor for Binance USDT-M Futures.',
                  style: TextStyle(color: ApexColors.textMuted),
                ),
                const SizedBox(height: 12),
                const Text(
                  'Use a Futures-only API key. Disable withdrawals; restrict to your IP. Trading risk is yours.',
                  style: TextStyle(color: ApexColors.highlight),
                ),
              ],
            ),
          ),
          const SizedBox(height: 12),
          TextField(
            controller: _key,
            onChanged: (_) => setState(() {}),
            decoration: const InputDecoration(labelText: 'API Key'),
          ),
          const SizedBox(height: 10),
          TextField(
            controller: _secret,
            obscureText: true,
            onChanged: (_) => setState(() {}),
            decoration: const InputDecoration(labelText: 'API Secret'),
          ),
          const SizedBox(height: 8),
          Row(
            children: [
              Switch(value: _testnet, onChanged: (v) => setState(() => _testnet = v)),
              const SizedBox(width: 8),
              const Text('Use Testnet', style: TextStyle(color: ApexColors.textMuted)),
            ],
          ),
          if (_error != null)
            Padding(
              padding: const EdgeInsets.only(top: 8),
              child: Text(_error!, style: const TextStyle(color: ApexColors.bear)),
            ),
          const SizedBox(height: 16),
          SizedBox(
            width: double.infinity,
            child: ElevatedButton(
              onPressed: canSave ? _save : null,
              child: Text(_saving ? 'Verifying…' : 'Verify & continue'),
            ),
          ),
        ],
      ),
    );
  }
}
